From 4a0115c6c877ca9f94ea187b5e64c8620bb0e74e Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Wed, 21 Dec 2022 18:42:18 +0100
Subject: [PATCH] Add several new functions from eremitein

Add several new functions from eremitein :
- Add safetynet, securize
- Add Autorun
- Add NoLog
- Add Restart sysui
- Add Dump Logs
- Add Disable Sound Volume Effect
---
 base.mk             |  4 ++-
 phh-on-data.sh      | 15 ++++++---
 phh-prop-handler.sh | 79 +++++++++++++++++++++++++++++++++++++++++++++
 rw-system.sh        |  3 ++
 secure.sh           | 16 +++++++++
 vndk.rc             | 18 +++++++++++
 6 files changed, 130 insertions(+), 5 deletions(-)
 create mode 100644 secure.sh

diff --git a/base.mk b/base.mk
index 965af89..12b1e88 100644
--- a/base.mk
+++ b/base.mk
@@ -51,7 +51,7 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 #Huawei HiSuite (also other OEM custom programs I guess) it's of no use in AOSP builds
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 	persist.sys.usb.config=adb \
-	ro.cust.cdrom=/dev/null	
+	ro.cust.cdrom=/dev/null
 
 #VNDK config files
 PRODUCT_COPY_FILES += \
@@ -87,6 +87,8 @@ PRODUCT_COPY_FILES += \
 	device/phh/treble/rw-system.sh:system/bin/rw-system.sh \
 	device/phh/treble/phh-on-data.sh:system/bin/phh-on-data.sh \
 	device/phh/treble/phh-prop-handler.sh:system/bin/phh-prop-handler.sh \
+	device/phh/treble/secure.sh:system/phh/secure.sh \
+	device/phh/treble/su/su:system/phh/xbin/su \
 	device/phh/treble/fixSPL/getSPL.arm:system/bin/getSPL
 
 PRODUCT_COPY_FILES += \
diff --git a/phh-on-data.sh b/phh-on-data.sh
index 472d16e..34ad54b 100644
--- a/phh-on-data.sh
+++ b/phh-on-data.sh
@@ -3,25 +3,32 @@
 vndk="$(getprop persist.sys.vndk)"
 [ -z "$vndk" ] && vndk="$(getprop ro.vndk.version |grep -oE '^[0-9]+')"
 
-if getprop persist.sys.phh.no_vendor_overlay |grep -q true;then
+if getprop persist.sys.phh.no_vendor_overlay |grep -q true; then
 	for part in odm vendor;do
 		mount /mnt/phh/empty_dir/ /$part/overlay
 	done
 fi
 
-if getprop persist.sys.phh.caf.media_profile |grep -q true;then
+if getprop persist.sys.phh.no_stock_apps |grep -q true; then
+	for part in odm vendor; do
+		mount /mnt/phh/empty_dir/ /$part/overlay
+		mount /mnt/phh/empty_dir/ /$part/app
+	done
+fi
+
+if getprop persist.sys.phh.caf.media_profile |grep -q true; then
     setprop media.settings.xml "/vendor/etc/media_profiles_vendor.xml"
 fi
 
 
 minijailSrc=/system/system_ext/apex/com.android.vndk.v28/lib/libminijail.so
 minijailSrc64=/system/system_ext/apex/com.android.vndk.v28/lib64/libminijail.so
-if [ "$vndk" = 27 ];then
+if [ "$vndk" = 27 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
 fi
 
-if [ "$vndk" = 28 ];then
+if [ "$vndk" = 28 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
     mount $minijailSrc64 /system/lib64/vndk-28/libminijail.so
diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 4371632..0afdd9b 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -195,6 +195,85 @@ if [ "$1" == "persist.sys.phh.backlight.scale" ];then
     exit
 fi
 
+if [ "$1" == "persist.sys.phh.qin.dt2w" ];then
+    if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
+        exit 1
+    fi
+
+    echo "$prop_value" > /sys/devices/platform/soc/soc:ap-apb/70800000.i2c/i2c-3/3-0038/fts_gesture_mode
+    exit
+fi
+
+#safetynet
+if [ "$1" == "persist.sys.phh.safetynet" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/secure ]; then
+        mkdir -p /data/adb/phh
+        cp /system/phh/secure.sh /data/adb/phh/secure
+    fi
+    /system/bin/sh /data/adb/phh/secure
+    exit
+fi
+
+#autorun
+if [ "$1" == "persist.sys.phh.autorun" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/run ]; then
+        mkdir -p /data/adb/phh
+        touch /data/adb/phh/run
+    fi
+    /system/bin/sh /data/adb/phh/run
+    exit
+fi
+
+#nolog
+if [ "$1" == "persist.sys.phh.nolog" ]; then
+    if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        setprop ctl.stop logd
+        setprop ctl.stop traced
+        setprop ctl.stop traced_probes
+    else
+        setprop ctl.start traced_probes
+        setprop ctl.start traced
+        setprop ctl.start logd
+    fi
+    exit
+fi
+
+#restart_sysui
+if [ "$1" == "sys.phh.restart_sysui" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        pkill systemui
+    fi
+    exit
+fi
+
+#dump_logs
+if [ "$1" == "sys.phh.dump_logs" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        logcat -b all -d > /sdcard/logs.txt
+    fi
+    exit
+fi
+
 if [ "$1" == "persist.sys.phh.disable_soundvolume_effect" ];then
     if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
         exit 1
diff --git a/rw-system.sh b/rw-system.sh
index 993b94b..2410ec7 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -728,7 +728,9 @@ copyprop() {
         resetprop_phh "$1" "$(getprop "$2")"
     fi
 }
+
 if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ];then
+
     copyprop ro.build.device ro.vendor.build.device
     copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
     copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
@@ -754,6 +756,7 @@ if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ];then
     copyprop ro.product.manufacturer ro.vendor.product.manufacturer
     copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
     copyprop ro.product.manufacturer ro.product.vendor.manufacturer
+
     (getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
         [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
     done
diff --git a/secure.sh b/secure.sh
new file mode 100644
index 0000000..98027e7
--- /dev/null
+++ b/secure.sh
@@ -0,0 +1,16 @@
+(getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
+    [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
+done
+resetprop_phh ro.build.tags release-keys
+resetprop_phh ro.boot.vbmeta.device_state locked
+resetprop_phh ro.boot.verifiedbootstate green
+resetprop_phh ro.boot.flash.locked 1
+resetprop_phh ro.boot.veritymode enforcing
+resetprop_phh ro.boot.warranty_bit 0
+resetprop_phh ro.warranty_bit 0
+resetprop_phh ro.debuggable 0
+resetprop_phh ro.secure 1
+resetprop_phh ro.build.type user
+resetprop_phh ro.build.selinux 0
+resetprop_phh ro.adb.secure 1
+setprop ctl.restart adbd
diff --git a/vndk.rc b/vndk.rc
index d1fffde..5e59a97 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -52,6 +52,9 @@ on property:persist.sys.phh.disable_soundvolume_effect=*
 on property:ro.sf.lcd_density=*
     restart surfaceflinger
 
+on property:persist.sys.phh.qin.dt2w=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.qin.dt2w"
+
 on property:persist.sys.phh.disable_a2dp_offload=0
     setprop persist.sys.phh.disable_a2dp_offload false
 
@@ -61,6 +64,21 @@ on property:persist.sys.phh.disable_a2dp_offload=1
 on property:persist.sys.phh.disable_a2dp_offload=*
     setprop persist.bluetooth.bluetooth_audio_hal.disabled ${persist.sys.phh.disable_a2dp_offload}
 
+on property:persist.sys.phh.safetynet=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.safetynet"
+
+on property:persist.sys.phh.autorun=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.autorun"
+
+on property:persist.sys.phh.nolog=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.nolog"
+
+on property:sys.phh.restart_sysui=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.restart_sysui"
+
+on property:sys.phh.dump_logs=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.dump_logs"
+
 on property:init.svc.ril-proxy=stopped && property:persist.sys.phh.restart_ril=true
     start ril-proxy
 
-- 
2.25.1

